subprojects/core/src/main/java/org/gradle/api/internal/StartParameterInternal.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal;
+
+import org.gradle.StartParameter;
+import org.gradle.api.Incubating;
+import org.gradle.util.DeprecationLogger;
+import org.gradle.util.SingleMessageLogger;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class StartParameterInternal extends StartParameter {
```

Comment:
Just as a side note: A pattern that we applied in the past for the `..Internal` classes was to introduce an interface instead e.g. `ProjectInternal`, `GradleInternal`. The interface would define the methods that should not be exposed to the public class. Too bad we can't use this pattern here has it would require us to turn `StartParameter` into an interface and create an implementation for it. It would be a breaking change so let's not do that.

Reply:
Thanks for your guidance.